### PR TITLE
docs: fsnotify: replace handmade diagram by mermaid diagram

### PR DESCRIPTION
**Pull request by alban**
Improvements from the previous diagram:
- The order of execution is more clear
- Show the alternatives (inotify vs fanotify)

```mermaid
sequenceDiagram
    box rgb(245, 245, 245) Programs
        participant Tracer
        participant Tracee
    end
    box rgb(245, 245, 245) Maps
        participant fsnotify_insert_event_ctx
        participant enriched_fsnotify_events
        participant fsnotify_remove_first_event_ctx
        participant events ring buffer
    end

alt inotify
    Note over Tracee: kprobe/inotify_handle_inode_event
    Activate Tracee
        rect rgb(255, 245, 168)
        Tracee ->> fsnotify_insert_event_ctx: insert into hash table<br/>Key: pid<br/>Value: {type=inotify, ino=123...}
        Activate fsnotify_insert_event_ctx
    end
else fanotify
    Note over Tracee: kprobe/fanotify_handle_event
    rect rgb(255, 245, 168)
        Tracee ->> fsnotify_insert_event_ctx: insert into hash table<br/>Key: pid<br/>Value: {type=fanotify, ino=123...}
    end
end

    Note over Tracee: kprobe/fsnotify_insert_event<br/>Input: struct fsnotify_group *group, struct fsnotify_event *event
    Activate Tracee
    rect rgba(255, 245, 168)
        Tracee ->> fsnotify_insert_event_ctx: lookup from hash table<br/>Key: pid<br/>Value: {type, ino...}
        Tracee ->> enriched_fsnotify_events: insert into hash table<br/>Key: fsnotify_event<br/>Value: {filename, tracee_comm, ...}
        Activate enriched_fsnotify_events
    end
    Deactivate Tracee

alt inotify
    Note over Tracee: kretprobe/inotify_handle_inode_event
    rect rgba(255, 245, 168)
        Tracee ->> fsnotify_insert_event_ctx: delete entry from hash table<br/>Key: pid
    end
else fanotify
    Note over Tracee: kretprobe/fanotify_handle_event
    rect rgba(255, 245, 168)
        Tracee ->> fsnotify_insert_event_ctx: delete entry from hash table<br/>Key: pid
        Deactivate fsnotify_insert_event_ctx
    end
end
    Deactivate Tracee

    Note over Tracer: kprobe/fsnotify_remove_first_event<br/>Input: struct fsnotify_group *group
    Activate Tracer
    rect rgba(255, 245, 168)
        Tracer ->> fsnotify_remove_first_event_ctx: insert into hash table<br/>Key: pid<br/>Value: fsnotify_group pointer
        Activate fsnotify_remove_first_event_ctx
    end

    Note over Tracer: kretprobe/fsnotify_remove_first_event<br/>Returns: struct fsnotify_event *event
    rect rgba(255, 245, 168)
        Tracer ->> fsnotify_remove_first_event_ctx: lookup+delete from hash table<br/>Key: pid
        Deactivate fsnotify_remove_first_event_ctx
        Tracer ->> enriched_fsnotify_events: lookup from hash table<br/>Key: fsnotify_event<br/>Value: {filename, tracee_comm, ...}
        Tracer ->> events ring buffer: submit event to ring buffer<br/>type = fanotify | inotify
        Note over events ring buffer: Inspektor Gadget<br/>receives the event
    end
    Deactivate Tracer

    Note over Tracee: kprobe/fsnotify_destroy_event<br/>Input: struct fsnotify_group *group, struct fsnotify_event *event
    Activate Tracee
    rect rgba(255, 245, 168)
        alt inotify or fanotify
        Note over events ring buffer: Nothing to send
        else fanotify_perm_response
        Tracee ->> events ring buffer: submit event to ring buffer<br/>type = fanotify_perm_response
        Note over events ring buffer: Inspektor Gadget<br/>receives the event
        end
        Tracee ->> enriched_fsnotify_events: delete from hash table<br/>Key: fsnotify_event
        Deactivate enriched_fsnotify_events
    end
    Deactivate Tracee
```

### CODE REVIEW

**Comment by blanquicet** [comment]
Mmmmm the website doesn't render the mermaid diagram 🙁:

image

Is there any way to do it?

File: gadgets/fsnotify/README.mdx [hunk 1]
```diff
@@ -114,4 +114,84 @@ ig           containerd-shim  fa_resp    FANOTIFY_EVENT_TYPE_PATH_PERM   allow
...
+        participant fsnotify_insert_event_ctx
+        participant enriched_fsnotify_events
+        participant fsnotify_remove_first_event_ctx
+        participant events ring buffer
+    end
+
+alt inotify
+    Note over Tracee: kprobe/inotify_handle_inode_event
+    Activate Tracee
+        rect rgb(255, 245, 168)
```
Comment (blanquicet) [inline gadgets/fsnotify/README.mdx:133]: IMO, the font color is not very visible over this background:

image

**Review by blanquicet — COMMENTED** [review]
> * The order of execution is more clear
> * Show the alternatives (inotify vs fanotify)
>
Definitely! 

I'm just wondering how to render it in the website.

**Comment by alban** [comment]
> Mmmmm the website doesn't render the mermaid diagram 🙁:

Mermaid graphs work fine on `docs/quick-start.md`: https://www.inspektor-gadget.io/docs/latest/quick-start

Is it because of the file extension `.md` vs `.mdx`?

File: gadgets/fsnotify/README.mdx (hunk 1 above)
Comment (alban) [reply gadgets/fsnotify/README.mdx:133]: How did you render it?

On [this page](https://github.com/inspektor-gadget/inspektor-gadget/blob/3aef50b3429515d1a59446036334d44d906e3746/gadgets/fsnotify/README.mdx), I see this:

image

**Comment by alban** [comment]
It works for me:

image


@blanquicet Do you have commit https://github.com/inspektor-gadget/website/pull/75 when you tested?

**Comment by blanquicet** [comment]
> @blanquicet Do you have commit [inspektor-gadget/website#75](https://github.com/inspektor-gadget/website/pull/75) when you tested?
>
Aaahh I was missing that, sorry.

File: gadgets/fsnotify/README.mdx (hunk 1 above)
Comment (blanquicet) [reply gadgets/fsnotify/README.mdx:133]: I think it's because I have dark theme on GH.

**Review by blanquicet — APPROVED** [review]
LGTM after finding a color that make text clearer when using dark theme.

**Comment by alban** [comment]
I couldn't find the perfect color but now it looks better:

image
image

**Pull request merged**